src/openforms/submissions/models.py

```diff
@@ -528,6 +532,29 @@ def payment_registered(self) -> bool:
         # TODO support partial payments
         return self.payments.filter(status=PaymentStatus.registered).exists()
 
+    def get_appointment_step_url(self) -> str:
+        for form_step in self.form.formstep_set.all():
+            for component in form_step.iter_components(recursive=True):
+                if component.get("appointments", {}).get("showProducts"):
+                    return form_step.form_definition.slug
+
+        return ""
+
+    @transaction.atomic
+    def copy(self):
+        copy = Submission.objects.create(
+            form=self.form,
+            form_url=self.form_url,
+            previous_submission=self,
+        )
+        for submission_step in self.submissionstep_set.all():
+            SubmissionStep.objects.create(
+                submission=copy,
+                form_step=submission_step.form_step,
+                data=submission_step.data,
+            )
+        return copy
```

Comment:
The canonical Django way is to grab the object, set `object.pk = None` and save it again. Then you're sure all the required fields are copied along.

In this case for example, we lose the authentication information and possible other meta-information.

Not sure how to best approach this :thinking: 

This would probably also be better suited as a model mananger function: `Submission.objects.copy(old_submission)`